### ISSUE TITLE
Keras: expose loss metric of singular output

### ISSUE THREAD

**Issue by phemmer** [tensorflow/tensorflow#36216]
**System information**
- TensorFlow version (you are using): 2.1.0
- Are you willing to contribute it (Yes/No): maybe

**Describe the feature and the current behavior/state.**
Currently when training a keras model with multiple outputs, you get multiple "loss" metrics. One for each output, and then an overall loss for the whole model. However when your model only contains a single output, the metrics only contain a single loss value. [The documentation](https://www.tensorflow.org/api_docs/python/tf/keras/Model?version=stable#compile) implies that this is because the overall loss is simply a weighted sum of the individual outputs, and thus is irrelevant when only one output is present. However this is incorrect. The overall loss used to train the model also includes other factors, such as [layer losses due to regularizer penalties](https://www.tensorflow.org/api_docs/python/tf/keras/regularizers/Regularizer). Thus even with a single output, the output's loss can be different from the model's overall loss.
To address this discrepancy, a metric for the output loss should be provided even when the model only has one output.

**Will this change the current api? How?**
No

**Who will benefit with this feature?**
Anyone with layer losses, such as due to regularizers.

**Any Other info.**

**Comment by phemmer** [comment]
As an example, I commented out [this line in `training.py`](https://github.com/tensorflow/tensorflow/blob/c044803bcd710a040d6baa6f39588581ae1c5c01/tensorflow/python/keras/engine/training.py#L413) to expose the output loss, and you can see that there is indeed a difference (my output is called `action`):

```
Epoch 19/40
167/168 [============================>.] - ETA: 0s - loss: 0.1629 - action_loss: 0.1886 - binary_crossentropy: 0.1405 - mean_absolute_error: 0.0486
168/168 [==============================] - 5s 32ms/step - loss: 0.1628 - action_loss: 0.1882 - binary_crossentropy: 0.1405 - mean_absolute_error: 0.0485 - val_loss: 0.0833 - val_action_loss: 0.1289 - val_binary_crossentropy: 0.1011 - val_mean_absolute_error: 0.0293
```

As you can see, `loss != action_loss` and `val_loss != val_action_loss`.

**Comment by phemmer** [comment]
I made the changes to implement the proposal, and they can be found here: https://github.com/tensorflow/tensorflow/compare/master...phemmer:keras-singular-output-loss

I have not created a pull request as I cannot get the tests working. Even the dockerized tests fail (unrelated to my change, some sort of bazel toolchain issue). The change is pretty small, so if someone else wants to take this and run with it, feel free.

**Comment by phemmer** [comment]
Updated changes for version 2.2: https://github.com/phemmer/tensorflow/commit/a588fcce4336ba4aa1e3856d9904debbcfe8d97d

**Comment by rmothukuru** [comment]
@phemmer,
Sorry for the delayed response. Can you please confirm if you have submitted a `Pull Request` with the changes you submitted in the above comment? Thanks!
